src/main/java/com/hedvig/contract/customerio/EventListenerStrategy.kt

```diff
@@ -0,0 +1,174 @@
+  ) {
+    val data = mapOf("activation_date" to contract.masterInception.toString())
+    notificationService.updateCustomer(contract.holderMemberId, data)
+  }
+
+  override fun onPendingAgreementActivated() {
+    val data = mapOf("activation_date" to contract.masterInception.toString())
+    notificationService.updateCustomer(contract.holderMemberId, data)
+  }
+
+  override fun onPendingAgreementReplaced() {
+    this.updateContractInfo(contract, sequenceNumber)
+  }
+
+  override fun onContractWithActiveAgreementCreated() {
+    this.updateContractInfo(contract, sequenceNumber)
+  }
+
+  override fun onContractWithPendingAgreementCreated() {
+    this.updateContractInfo(contract, sequenceNumber)
+  }
+
+  override fun onAgreementAddedWithoutPredecessorAdjusted(
+    contract: ContractState<AgreementState>,
+    sequenceNumber: Long
+  ) {
+    this.updateContractInfo(contract, sequenceNumber)
+  }
+
+  override fun onAgreementAddedWithPredecessorAdjusted(contract: ContractState<AgreementState>, sequenceNumber: Long) {
+    this.updateContractInfo(contract, sequenceNumber)
+  }
+
+  override fun onAgreementReplaced(contract: ContractState<AgreementState>, sequenceNumber: Long) {
+    this.updateContractInfo(contract, sequenceNumber)
+  }
+
+  private fun extractData(
+    agreement: SwedishApartmentAgreementState,
+    masterInceptionDate: LocalDate? = null
+  ): Map<String, Any?> {
+    return mapOf(
+      "activation_date" to (masterInceptionDate ?: agreement.from)?.toString(),
+      "insurance_address_street" to agreement.address.street,
+      "insurance_address_city" to agreement.address.city,
+      "insurance_address_zip" to agreement.address.postalCode,
+      "household_size" to agreement.numberCoInsured + 1,
+      "monthly_subscription" to agreement.basePremium,
+      "is_student" to agreement.lineOfBusiness.isStudent,
+      "insurance_type" to agreement.lineOfBusiness.name
+    )
+  }
+
+  private fun extractData(
+    agreement: SwedishHouseAgreementState,
+    masterInceptionDate: LocalDate? = null
+  ): Map<String, Any?> {
+    return mapOf(
+      "activation_date" to (masterInceptionDate ?: agreement.from)?.toString(),
+      "insurance_address_street" to agreement.address.street,
+      "insurance_address_city" to agreement.address.city,
+      "insurance_address_zip" to agreement.address.postalCode,
+      "household_size" to agreement.numberCoInsured + 1,
+      "monthly_subscription" to agreement.basePremium,
+      "is_student" to null,
+      "insurance_type" to "HOUSE"
+    )
+  }
+}
+
+class Norway(
+  private val notificationService: NotificationService,
+  private val contract: ContractState<AgreementState>,
+  private val sequenceNumber: Long?
+) : EventListenerStrategy {
+  private fun execWithCurrentOrLastContract(
+    contract: ContractState<AgreementState>,
+    sequenceNumber: Long?
+  ) {
+    val possibleMasterInceptionDate = if (!contract.hasPendingAgreement) contract.masterInception else null
```

Comment:
There is a more explicit function called
```suggestion
    val possibleMasterInceptionDate = if (contract.hasMasterInception) contract.masterInception else null
```